### PR TITLE
debounce resize event listener for mutation observer

### PR DESCRIPTION
**Pull request by webb04**
## Why are you doing this?

There's a heavy `stringify` operation inside the `callback` function that should be debounced inside the event listener.

### CODE REVIEW

File: src/client/nativeCommunication.ts [hunk 1]
```diff
@@ -62,7 +62,15 @@ function insertAds(): void {
...
 
-        window.addEventListener('resize', callback);
+
+        let timeout: number;
+        window.addEventListener('resize', () => {
+            if (timeout) {
+                window.cancelAnimationFrame(timeout);
+            }
+            timeout = window.requestAnimationFrame(callback);
+        }, false);
```
Comment (JamieB-gu) [inline src/client/nativeCommunication.ts:72]: Would you be able to give a brief explanation of how this works?

Comment (webb04) [reply src/client/nativeCommunication.ts:72]: https://gomakethings.com/debouncing-your-javascript-events/#requestanimationframe

Comment (JamieB-gu) [reply src/client/nativeCommunication.ts:72]: Under what circumstances will the `timeout` in `if (timeout)` be falsy?

Comment (webb04) [reply src/client/nativeCommunication.ts]: The first time when there is no previous animation frame to cancel. I've renamed it as it's not really a "timeout"

File: src/client/nativeCommunication.ts [hunk 2]
```diff
@@ -183,7 +183,13 @@ function reportNativeElementPositionChanges(): void {
         }
     };
 
-    window.addEventListener('resize', callback);
+    let currentAnimationFrame: number;
+    window.addEventListener('resize', () => {
+        if (currentAnimationFrame) {
```
Comment (JamieB-gu) [inline src/client/nativeCommunication.ts:188]:
```suggestion
    let currentAnimationFrame: number | null = null;
    window.addEventListener('resize', () => {
        if (currentAnimationFrame !== null) {
```

**Pull request merged**